pkg/cmd/pulumi/state/state_protect.go

```diff
@@ -113,7 +113,10 @@ func protectAllResources(
 			}
 
 			for _, res := range snap.Resources {
-				res.Protect = true
+				// Skip resources that are pending deletion
+				if !res.Delete {
```

Comment:
Do resources that are pending replacement need the same treatment? (I don't know the answer, happy to have that in a separate PR and get this merged).